cores/nRF5/WInterrupts.c

```diff
@@ -95,24 +96,67 @@ int attachInterrupt(uint32_t pin, voidFuncPtr callback, uint32_t mode)
       return 0;
   }
 
+  // All information for the configuration is known, except the prior values
+  // of the config register.  Pre-compute the mask and new bits for later use.
+  //     CONFIG[n] = (CONFIG[n] & oldRegMask) | newRegBits;
+  //
+  // Three fields are configured here: PORT/PIN, POLARITY, MODE
+  const uint32_t oldRegMask = ~(GPIOTE_CONFIG_PORT_PIN_Msk | GPIOTE_CONFIG_POLARITY_Msk | GPIOTE_CONFIG_MODE_Msk);
+  const uint32_t newRegBits =
+    ((pin                      << GPIOTE_CONFIG_PSEL_Pos    ) & GPIOTE_CONFIG_PORT_PIN_Msk) |
+    ((polarity                 << GPIOTE_CONFIG_POLARITY_Pos) & GPIOTE_CONFIG_POLARITY_Msk) |
+    ((GPIOTE_CONFIG_MODE_Event << GPIOTE_CONFIG_MODE_Pos    ) & GPIOTE_CONFIG_MODE_Msk    ) ;
+
+  // To avoid unexpected consequences, first loop through the channelMap
+  // to preferably re-use any channel that was already in use (even if
+  // earlier channel is no longer in use).
   for (int ch = 0; ch < NUMBER_OF_GPIO_TE; ch++) {
```

Comment:
two loops approach is good, but It duplicates the actual handling code. I would be better to move the 2 loop() into a seperate find_available_channel() function, that
- return currently used channel if any else
- return first empty
- return (-1) if non

Then we will only need to write handling code once.

Reply:
Well, the handling code is **_not_** common (and should not be common) between the two cases.  Still, I am updating per your request.